refactor(automation): Simplify docker compose files

Cliftonz opened the pull request:
This PR removes old and unused docker compose files and gives more flexibility to the base docker compose file by allowing the user to select the port of the infrastructure.

Cliftonz committed:
Remove redis cluster docker files

Cliftonz committed:
Remove unused Docker Compose files

These Docker Compose files are no longer in use by the application. It is beneficial to maintain clarity and cleanliness of the repository by removing them.

Cliftonz committed:
Update docker compose and environment files for development setup

The docker compose file for local development setup now utilizes environment variables for port numbers. This makes it easier for developers to use custom port numbers without modifying the docker-compose file directly. Moreover, a readme file was updated to provide instructions for setting up the local development environment. The name of the docker compose file was also changed for consistency.

Cliftonz committed:
Move docker-compose.monitoring.yml to deployment directory

The docker-compose file specific for monitoring has been relocated from the development directory to the deployment directory. This reorganization reflects its purpose more accurately.

Cliftonz committed:
Merge branch 'refs/heads/next' into update-containized-env

SokratisVidros commented on docker/local/deployment/docker-compose.monitoring.yml:
Should we merge the two docker-compose files into one? The `monitoring`, `local` suffixes are a bit misleading. We might prefer to use `docker-compose up SERVICE1 SERVICE2 ...` and target a single `docker-compose.yml`.

Cliftonz replied:
I think its fine to combine and remove all of the other files. I think there was alot of bloat in here and realistically we can make the sub systems profiles.

SokratisVidros commented on docker/local/development/docker-compose.yml:
```diff
@@ -8,7 +8,7 @@ services:
       - SERVICES=s3
     ports:
-      - "4566:4566"
+      - "${DOCKER_LOCALSTACK_PORT}:4566"
```
```suggestion
      - "${DOCKER_LOCALSTACK_PORT:-4566}:4566"
```

SokratisVidros replied:
I am not a 100% sure if this works but it's a nice touch to add overridable, default values.

SokratisVidros commented on docker/local/development/docker-compose.yml:
```diff
@@ -21,7 +21,7 @@ services:
     network_mode: bridge
     ports:
-      - "27017:27017"
+      - "${DOCKER_MONGO_PORT}:27017"
```
```suggestion
      - "${DOCKER_MONGO_PORT:-27017}:27017"
```

SokratisVidros commented on docker/local/development/docker-compose.yml:
```diff
@@ -33,7 +33,7 @@ services:
     network_mode: bridge
     ports:
-      - "6379:6379"
+      - "${DOCKER_REDIS_SERVICE_PORT}:6379"
```
```suggestion
      - "${DOCKER_REDIS_SERVICE_PORT:-6379}:6379"
```

Cliftonz committed:
Update docker/local/development/docker-compose.yml

Co-authored-by: Sokratis Vidros <SokratisVidros@users.noreply.github.com>

Cliftonz committed:
Update docker/local/development/docker-compose.yml

Co-authored-by: Sokratis Vidros <SokratisVidros@users.noreply.github.com>

Cliftonz committed:
Update docker/local/development/docker-compose.yml

Co-authored-by: Sokratis Vidros <SokratisVidros@users.noreply.github.com>

Cliftonz committed:
Refactor and clean up Docker configuration files

This commit streamlines Docker configuration files by consolidating and removing redundant docker-compose.yml from specific subdirectories. Also, the README.md files have been updated to reflect the changes made to the .env file path and docker-compose command. Necessary information about running Docker in ARM only mode has been added to the enterprise Readme file.

Cliftonz committed:
Merge remote-tracking branch 'origin/update-containized-env' into update-containized-env

Cliftonz committed:
Merge branch 'next' into update-containized-env

Cliftonz committed:
Merge branch 'next' into update-containized-env
